sms-android/snippet.html

```diff
@@ -0,0 +1,240 @@
+   .button:link:active,
+   .button:visited:active,
+   .button.active,
+   .button:link.active,
+   .button:visited.active {
+       box-shadow: inset 0px 2px 0px 0px rgba(0, 0, 0, 0.2), inset 0px 12px 24px 6px rgba(0, 0, 0, 0.2), inset 0px 0px 2px 2px rgba(0, 0, 0, 0.2);
+       color: #ffffff;
+       text-decoration: none;
+       -webkit-transition: all linear 0.25s;
+       transition: all linear 0.25s;
+   }
+
+   #firefox-android-sms-{{ snippet_id }} .button.insensitive,
+   .button:link.insensitive,
+   .button:visited.insensitive,
+   .button.insensitive:hover,
+   .button:link.insensitive:hover,
+   .button:visited.insensitive:hover,
+   .button.insensitive:focus,
+   .button:link.insensitive:focus,
+   .button:visited.insensitive:focus {
+       background-color: #9ca3aa;
+       background-repeat: repeat-x;
+       background-image: -webkit-linear-gradient(top, #bfc7cd, #9ca3aa);
+       /* Safari 5.1+, Chrome 10+ */
+       -ms-filter: "progid:DXImageTransform.Microsoft.gradient(startColorstr='#bfc7cd', endColorstr='#9ca3aa', GradientType=0)";
+       /* IE8+ */
+       background-image: linear-gradient(to bottom, #bfc7cd, #9ca3aa);
+       /* the standard */
+       box-shadow: 0px 1px 0px 0px rgba(0, 0, 0, 0.2), inset 0px -1px 0px 0px rgba(0, 0, 0, 0.3);
+       color: #eee;
+   }
+
+   #firefox-android-sms-{{ snippet_id }} .button.arrow span {
+       display: block;
+       margin-right: -12px;
+       padding: 0 30px 0 0;
+       background: url(https://www.mozilla.org/media/img/sandstone/buttons/arrow-go-r.png) right center no-repeat;
+   }
+
+   #firefox-android-sms-{{ snippet_id }} #number {
+       height: 25px;
+       width: 180px;
+       background: #fff;
+       border-color: #b2b2b2;
+       border-style: solid;
+       border-width: 1px;
+       padding: 3.3333333333333335px 6.666666666666667px;
+       border-radius: 3px;
+       box-shadow: 0 1px rgba(255, 255, 255, 0.5);
+   }
+  </style>
+  <script type="text/javascript">
+   //<![CDATA[
+   (function() {
+       var snippet = document.getElementById('firefox-android-sms-{{ snippet_id }}');
+       snippet.addEventListener('show_snippet', function() {
+           var input = document.querySelector('#firefox-android-sms-{{ snippet_id }} #number');
+           input.addEventListener('input', function() {
+               var button = document.querySelector('#firefox-android-sms-{{ snippet_id }} #submit-button');
+               if (input.value.length >= 10) {
+                   button.disabled = false;
+                   button.classList.remove('insensitive');
+               }
+               else {
+                   button.disabled = true;
+                   button.classList.add('insensitive');
+               }
+           });
+           var submit = document.getElementById('submit-button');
+           var form = document.getElementById('snippet-{{ snippet_id }}-sms-send');
+           submit.addEventListener('click', function(event) {
```

Comment:
Directly == 'by hitting enter while text input is focused'?

Currently the event triggers both on button click and on enter.